lib/mayaUsd/fileio/primUpdaterManager.h

```diff
@@ -128,11 +128,6 @@ class PrimUpdaterManager : public PXR_NS::TfWeakBase
 
     bool _inPushPull { false };
 
-    // Becomes true when there is at least one pulled prim.
-    // The goal is to let code that can be optimized when there is no pull prim
-    // to check rapidly.
```

Comment:
What about this optimization? Now you are using findPullRoot() which searches thru the entire dag hierarchy looking for one.

Reply:
It does search the entire DAG.

Only the direct children of the root are iterated. Note that we do that find operation whenever we actually edit-as-Maya or any other edit-related op, so it surely is not slow.

If we do find it became a bottleneck, we can even optimize it further by using a MSelection with the pull-set name, which would be even faster.

Reply:
I changed findPullRoot() to use MSelectionList, so now there is no iteration at all, just direct access to the object, so there should be no performance risk. Bonus: the code is also faster for the actual operations.

Reply:
Great that is much better.